### ISSUE TITLE
Roslyn: How to get all projects from a solution (old vs. sdk style csproj)

### ISSUE THREAD

**Issue by sschmid** [sschmid/Entitas-CSharp#994]
Hi!

I'm having an Roslyn issue, sounds simple, but can't get it to work:

Task:

- get all projects in a solution and iterate over their documents

Problem:

- sdk-style csproj have no documents, only the old-style csproj

I created a test repo to test this in isolation
https://github.com/sschmid/RoslynTest

This will be very useful for the Entitas code generator.

Does anyone how to fix this?

**Comment by JesseTG** [comment]
I don't know how _exactly_ this is solved, but I know that [Genesis](https://github.com/jeffcampbellmakesgames/Genesis) has solved it. [This is the directory you'll want, I think.](https://github.com/jeffcampbellmakesgames/Genesis/tree/develop/ExternalApp/Genesis) Does that help?

**Comment by sschmid** [comment]
Yeah, he's using `MSBuildLocator.RegisterDefaults()` from `Microsoft.Build.Locator`.
Unfortunately:
> Operation is not supported on this platform

Only works on Windows

I'm using mono-hosted-msbuild
https://github.com/dotnet/msbuild/releases/mono-hosted-msbuild-v0.03

to be able to run roslyn on macOS and linux

**Comment by sschmid** [comment]
But I was about to try that with GitHub actions on a windows hosted machine. Let's see if that works

**Comment by sschmid** [comment]
I will move discussion to:
https://github.com/sschmid/RoslynTest/issues/1

**Comment by JesseTG** [comment]
Hey @jeffcampbellmakesgames, what do you think?

**Comment by sschmid** [comment]
Solved.

Entitas and Desperate devs was .NET 4.x to be compatible with Unity.

Jenny however can be net5.0 or newer.
Using net6.0 fixes it 😍
